core/translate/update.rs

```diff
@@ -188,13 +193,98 @@ pub fn prepare_update_plan(
             .collect()
     });
-    // Parse the WHERE clause
-    parse_where(
-        body.where_clause.as_ref().map(|w| *w.clone()),
-        &mut table_references,
-        Some(&result_columns),
-        &mut where_clause,
-    )?;
+
+    // Sqlite determines we should create an ephemeral table if we do not have a FROM clause
+    // Difficult to say what items from the plan can be checked for this so currently just checking if a RowId Alias is referenced
+    // https://github.com/sqlite/sqlite/blob/master/src/update.c#L395
+    // https://github.com/sqlite/sqlite/blob/master/src/update.c#L670
+    let columns = table.columns();
+
+    let rowid_alias_used = set_clauses.iter().fold(false, |accum, (idx, _)| {
+        accum || columns[*idx].is_rowid_alias
+    });
+
+    let (ephemeral_plan, where_clause) = if rowid_alias_used {
+        let internal_id = program.table_reference_counter.next();
+
+        let joined_tables = vec![JoinedTable {
+            table: match table.as_ref() {
+                Table::Virtual(vtab) => Table::Virtual(vtab.clone()),
+                Table::BTree(btree_table) => Table::BTree(btree_table.clone()),
+                _ => unreachable!(),
+            },
+            identifier: table_name.0.clone(),
+            internal_id,
+            op: Operation::Scan {
+                iter_dir,
+                index: None,
+            },
+            join_info: None,
+            col_used_mask: ColumnUsedMask::new(),
+        }];
+        let mut table_references = TableReferences::new(joined_tables, vec![]);
+
+        // Parse the WHERE clause
+        parse_where(
+            body.where_clause.as_ref().map(|w| *w.clone()),
+            &mut table_references,
+            Some(&result_columns),
+            &mut where_clause,
+        )?;
+
+        let index = Arc::new(Index {
+            name: "ephemeral_scratch".to_string(),
+            table_name: table_name.0.clone(),
+            ephemeral: true,
+            root_page: 0,
+            columns: vec![],
+            unique: true,
+            has_rowid: true,
+        });
+
+        let temp_cursor_id = program.alloc_cursor_id(CursorType::BTreeIndex(index.clone()));
+
+        program.emit_insn(Insn::OpenEphemeral {
+            cursor_id: temp_cursor_id,
+            is_table: false,
+        });
+
+        let ephemeral_plan = SelectPlan {
+            table_references,
+            result_columns: vec![ResultSetColumn {
+                expr: Expr::RowId {
+                    database: None,
+                    table: internal_id,
+                },
+                alias: None,
+                contains_aggregates: false,
+            }],
+            where_clause,       // original WHERE terms from the UPDATE clause
+            group_by: None,     // N/A
+            order_by: None,     // N/A
+            aggregates: vec![], // N/A
+            limit: None,        // N/A
+            query_destination: QueryDestination::EphemeralIndex {
```

Comment:
How about a new `QueryDestination::EphemeralTable` variant, so that we do the same as SQLite does?